move_base/src/move_base.cpp

```diff
@@ -1200,4 +1207,19 @@ namespace move_base {
 
     return true;
   }
+
+  // publish in context of action feedback, path left and time left to the goal
+  void MoveBase::publishPathMsg(const geometry_msgs::PoseStamped& current_position, const geometry_msgs::PoseStamped& goal)
+  {
+    move_base::Path msg;
+    msg.current_pose = current_position;
+    msg.goal_pose = goal;
+    msg.dist_to_goal = distance(current_position, goal);
+    geometry_msgs::Twist cmd_vel;
+    tc_->computeVelocityCommands(cmd_vel);
```

Comment:
This function is quite heavy and result is veeery variable, use max speed.

Reply:
Ok. What is the class member of max_speed?

Reply:
Have no idea.

Reply:
Thanks a lot.

Reply:
Or expose it to move_base somehow.